### ISSUE TITLE
NavigationViewController nonfunctional and can crash when initialized in storyboard

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#2973]
The workflow in the “[Storyboards](https://docs.mapbox.com/ios/navigation/guides/storyboards/)” guide doesn’t work because `NavigationViewController(coder:)` doesn’t call either `NavigationViewController(for:routeIndex:routeOptions:navigationOptions:)` or `NavigationViewController(for:routeIndex:routeOptions:navigationService:)`. Over time, `NavigationViewController(for:routeIndex:routeOptions:navigationOptions:)` has taken on some important responsibilities like setting up the speech controller and navigation service; without calling into it, the view controller would be nonfunctional. Additionally, if the application sets properties like `indexedRoute` and `navigationOptions` when segueing, as recommended by the guide, it would immediately crash because of setters that depend on implicitly unwrapped optionals (under the assumption that the other initializer has already been called).

`NavigationViewController(coder:)` would differ subtly from `NavigationViewController(for:routeIndex:routeOptions:navigationOptions:)` in that it needs to assume `navigationService` has already been initialized and set.

/ref #1063
/cc @mapbox/navigation-ios

**Comment by 1ec5** [comment]
The implementation could look something like the implementation of `NavigationViewController(for:routeIndex:routeOptions:navigationService:)`:

```swift
required public init?(coder aDecoder: NSCoder) {
    super.init(coder: aDecoder)
    let options = NavigationOptions(navigationService: navigationService)
    self.init(for: indexedRoute.0, routeIndex: indexedRoute.1, routeOptions: routeOptions, navigationOptions: options)
}
```

**Comment by MaximAlien** [comment]
Fixed in #2974.
